go/connector-errors/errors.go

```diff
@@ -0,0 +1,57 @@
+package errors
+
+import (
+	"errors"
+	"os"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// UserError wraps a source error with a user-facing message for the error string. The source error
+// can be provided so that it can be logged separately from the user-facing message for diagnostic
+// purposes.
+type UserError struct {
+	message string
+	source  error
+}
+
+// NewUserError creates a UserError that will output message as the error string.
+func NewUserError(message string, source error) *UserError {
```

Comment:
Typical style for this sort of error-wrapping function in Go would be `NewUserError(err, "this is a message")` so that the short, almost-always-identical prefix can be gotten out of the way first.